AutomatedTesting/Gem/PythonTests/scripting/ScriptEvents_AllParamDatatypes_CreationSuccess.py

```diff
@@ -5,170 +5,65 @@
-    file_saved          = ("Script event file saved",     "Script event file did not save")
-    node_found          = ("Node found in Script Canvas", "Node not found in Script Canvas")
-# fmt: on
-
-
-FILE_PATH = os.path.join(paths.projectroot, "TestAssets", "test_file.scriptevents")
 N_VAR_TYPES = 10  # Top 10 variable types
-SEARCH_RETRY_ATTEMPTS = 20
-TEST_METHOD_NAME = "test_method_name"
 
-class TestScriptEvents_AllParamDatatypes_CreationSuccess():
-    """
-    Summary:
-     Parameters of all types can be created.
-
-    Expected Behavior:
-     The Method handles the large number of Parameters gracefully.
-     Parameters of all data types can be successfully created.
-     Updated ScriptEvent toast appears in Script Canvas.
-
-    Test Steps:
-     1) Open Asset Editor
-     2) Initially create new Script Event file with one method
-     3) Add new method and set name to it
-     4) Add new parameters of each type
-     5) Verify if parameters are added
-     6) Expand the parameter rows
-     7) Set different names and datatypes for each parameter
-     8) Save file and verify node in SC Node Palette
-     9) Close Asset Editor
-
-    Note:
-     - This test file must be called from the Open 3D Engine Editor command terminal
-     - Any passed and failed tests are written to the Editor.log file.
-            Parsing the file or running a log_monitor are required to observe the test results.
-
-    :return: None
+def TestScriptEvents_AllParamDatatypes_CreationSuccess():
     """
+       Summary:
+        Parameters of all types can be created.
+
+       Expected Behavior:
+        The Method handles the large number of Parameters gracefully.
+        Parameters of all data types can be successfully created.
+        Updated ScriptEvent toast appears in Script Canvas.
+
+       Test Steps:
+        1) Get a handle on our O3DE QtPy objects then initialize the Asset Editor object
+        2) Create new Script Event Asset
+        3) Add a method to the script event
+        4) Add new parameters of each type
+        5) Close Asset Editor
+
+       Note:
+        - Any passed and failed tests are written to the Editor.log file.
+               Parsing the file or running a log_monitor are required to observe the test results.
+
+       :return: None
+       """
+    import azlmbr.legacy.general as general
+    from editor_python_test_tools.QtPy.QtPyO3DEEditor import QtPyO3DEEditor
+    from scripting_utils.scripting_constants import (SCRIPT_CANVAS_UI, ASSET_EDITOR_UI)
+
+    # Preconditions
+    general.idle_enable(True)
+
+    # 1) Get a handle on our O3DE QtPy objects then initialize the Asset Editor object
+    qtpy_o3de_editor = QtPyO3DEEditor()
+    # Close and reopen Asset Editor to ensure we don't have any existing assets open
+    qtpy_o3de_editor.close_asset_editor()
+    qtpy_asset_editor = qtpy_o3de_editor.open_asset_editor()
+
+    # 2) Create new Script Event Asset
+    qtpy_asset_editor.click_menu_bar_option("Script Events")
```

Comment:
Use scripting/asset editor consts.

Reply:
imported the constant and replaced the hard coded string